src/main/java/com/csfrequency/firebase/devicestore/FirebaseDeviceStore.java

```diff
@@ -111,6 +111,11 @@ public void onComplete(@NonNull Task<InstanceIdResult> task) {
         authStateListener = new FirebaseAuth.AuthStateListener() {
             @Override
             public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
+
+                // TODO: Are we guranteed that this callback fires before the InstanceId listener
+                // above fires? If the instance ID fires before we assign the currentUserm then we
+                // will never set the token.
+
```

Comment:
I've tweaked this slightly as part of the PR.

Essentially, whichever finishes second out of the instance ID listener and the auth listener will result in Firestore being updated.